src/amp-story-player/amp-story-player-impl.js

```diff
@@ -309,9 +315,25 @@ export class AmpStoryPlayer {
     this.initializeIframes_();
     this.initializeButton_();
     this.signalReady_();
+
+    if (this.isAutoplaying_) {
+      this.setUpAutoplayClickShield_();
+    }
+
     this.isBuilt_ = true;
   }
 
+  /**
+   * Sets up a listener disable autoplay mode on
+   * the first user interaction.
+   * @private
+   */
+  setUpAutoplayClickShield_() {
+    this.element_.addEventListener('click', () =>
+      this.toggleAutoplayMode_(false)
```

Comment:
This code will run on every click within the player. Should we unlisten when autoplay is off, and listen again when autoplay is on?
There would be no way to set it back on otherwise, as this would set it back to off.

Reply:
Given that the iframe will consume the click when the "click shield" is not applied, I don't think it's necessary to detach/attach this listener on every toggle, which is why I recommended doing it this way. But unlistening/listening on toggle sgtm!